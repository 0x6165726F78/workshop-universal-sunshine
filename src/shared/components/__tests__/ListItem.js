/**
 * @flow
 */
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('ListItem', () => {
  it('renders properly', () => {
    // eslint-disable-next-line global-require
    const ListItem = require('../ListItem').default;

    // Some random data
    const item = {
      date: '2018-01-22',
      date_epoch: 1516579200,
      day: {
        maxtemp_c: 18.2,
        maxtemp_f: 64.8,
        mintemp_c: 2.2,
        mintemp_f: 36.0,
        avgtemp_c: 9.6,
        avgtemp_f: 49.2,
        maxwind_mph: 11.9,
        maxwind_kph: 19.1,
        totalprecip_mm: 0.0,
        totalprecip_in: 0.0,
        avgvis_km: 20.0,
        avgvis_miles: 12.0,
        avghumidity: 36.0,
        condition: {
          text: 'Partly cloudy',
          icon: '//cdn.apixu.com/weather/64x64/day/116.png',
          code: 1003,
        },
        uv: 39960.0,
      },
      astro: {
        sunrise: '07:29 AM',
        sunset: '06:29 PM',
        moonrise: '11:10 AM',
        moonset: '11:30 PM',
      },
    };

    const tree = renderer.create(<ListItem item={item} />);

    expect(tree).toMatchSnapshot();
  });
});
