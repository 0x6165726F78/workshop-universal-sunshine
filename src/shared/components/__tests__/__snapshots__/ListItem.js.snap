// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders properly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingRight": 8,
          },
          Object {
            "flexShrink": 1,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/shared/icons/ic_light_clouds.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 36,
            "marginRight": 16,
            "width": 36,
          }
        }
      />
      <View
        style={
          Object {
            "flexShrink": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            }
          }
        >
          Sunday
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
            }
          }
        >
          Partly cloudy
        </Text>
      </View>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          }
        }
      >
        65 °F
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
          }
        }
      >
        36 °F
      </Text>
    </View>
  </View>
</View>
`;
